lib/kubeclient/watch_stream.rb

```diff
@@ -29,6 +29,8 @@ def each
         end
       rescue IOError
         raise unless @finished
+      ensure
+        finish
```

Comment:
@holyketzer should we now protect the `@http_client.close` call below in case it happens on an already closed connection?